3.10/aql/operations-replace.md

```diff
@@ -25,104 +25,171 @@ REPLACE <em>keyExpression</em> WITH <em>document</em> IN <em>collection</em></co
 Both variants can optionally end with an `OPTIONS { … }` clause.
 
 `collection` must contain the name of the collection in which the documents should
-be replaced. `document` is the replacement document. When using the first syntax, `document` 
-must also contain the `_key` attribute to identify the document to be replaced. 
+be replaced. `document` must be an object that contains the attributes and values 
```

Comment:
Only in the second variant. In the first one, `document` just provides the `_key` attribute (as on the second variant).

BWT, I'd vote to stress (both here and in the description of `UPDATE`) that the `document` attribute of the first variant is, in fact, just another way to describe `_key`. Otherwise the use of `document` is rather counter-intuitive and even misleading: it has nothing to do with the replacing `document` (except for the value of `_key` which is implicit in the `REPLACE`/`UPDATE`call).

Reply:
> Only in the second variant

I don't follow. With both syntaxes, `document` must be an object. This is not optional. If you use the first syntax, then this object must have a `_key` attribute. Other than that, the object can be empty, but you do commonly specify attributes - or you end up with an empty document (except system attributes).

Reply:
> > Only in the second variant
> 
> I don't follow. With both syntaxes, `document` must be an object. This is not optional. If you use the first syntax, then this object must have a `_key` attribute. Other than that, the object can be empty, but you do commonly specify attributes - or you end up with an empty document (except system attributes).

It was my mistake, sorry for the confusion.